overlord/snapstate/autorefresh.go

```diff
@@ -407,6 +413,38 @@ func canContinueAutoRefresh(st *state.State) (int, bool) {
 	return 0, false
 }
 
+func (m *autoRefresh) restoreMonitoring() error {
+	if m.restoredMonitoring {
+		return nil
+	}
+
+	var monitored []string
+	if err := m.state.Get("monitored-snaps", &monitored); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+
+	defer func() { m.restoredMonitoring = true }()
+	if len(monitored) == 0 {
+		return nil
+	}
+
+	monitoring := make(map[string]chan<- bool)
+	for _, snap := range monitored {
+		done := make(chan string, 1)
+		if err := cgroupMonitorSnapEnded(snap, done); err != nil {
+			return fmt.Errorf("cannot monitor for snap closure: %w", err)
+		}
+
+		abort := make(chan bool, 1)
+		monitoring[snap] = abort
+
+		go continueRefreshOnSnapClose(m.state, snap, done, abort)
```

Comment:
Just want to double check the placement `continueRefreshOnSnapClose` in this loop with error, before updating of the monitoring state (also different to single snap case in `asyncRefreshOnSnapClose`). It seems like there is risk for monitoring state that is not reflecting the actual monitoring activity? Also `continueRefreshOnSnapClose` also depends on/access monitoring state.